rtwo/drivers/openstack_network.py

```diff
@@ -148,19 +148,24 @@ def disassociate_floating_ip(self, server_id):
         deleted_ip = self.neutron.delete_floatingip(floating_ip_id)
         return
 
-    def associate_floating_ip(self, server_id):
+    def associate_floating_ip(self, server_id, external_network_id=None):
         """
         Create a floating IP on the external network
         Find port of new VM
         Associate new floating IP with the port assigned to the new VM
+        If external_network_id is not specified then an external network is
+        chosen arbitrarily. This may not work if there are multiple external
+        networks.
         """
         networks = self.list_networks()
-        external_networks = [net for net in networks 
-                             if net['router:external'] == True]
-        if not external_networks:
-            raise Exception("CONFIGURATION ERROR! No external networks found!"
-                            " Cannot associate floating ip without it!"
-                            " Create a fixed IP/port first!")
+        if external_network_id is None:
+            external_networks = [net for net in networks
+                                 if net['router:external'] == True]
```

Comment:
This is another example of a truthy claim that can be cleaned up. You didn't respond to my first comment. I'm guessing it feels more intuitive for you to see `== True`.

I grew out of this and would read the following as `all networks where router is external`
```
[net for net in networks if net['router:external']]
```

In languages like elm, it makes even less sense to do `== True` because the value is literally true or false, and so it is purely redundant. Or its not a boolean and a type error.

Reply:
Yes, `== True` reads more clearly to me, but I'm happy to do what is more terse/idiomatic.

Reply:
These lines pre-date this PR and I was going to leave them alone, but happy to change as part of this.